### PR TITLE
Optimize GitHub Actions Workflow for Android CI

### PR DESCRIPTION
**Pull request by Basler182**
# *Optimize GitHub Actions Workflow for Android CI*

## :recycle: Current situation & Problem
*Link any open issues or pull requests (PRs) related to this PR. Please ensure that all non-trivial PRs are first tracked and discussed in an existing GitHub issue or discussion.*

## :books: Documentation

- add google-services.json in workflow 
- allow manual triggering of workflow

## :pencil: Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Review by PSchmiedmayer — APPROVED** [review]
Thank you @Basler182; I added some additional context about the secrets in the Slack discussion.

File: .github/workflows/android.yml [hunk 1]
```diff
@@ -24,6 +26,11 @@ jobs:
           distribution: 'temurin'
           cache: gradle
 
+      - name: Decode and write google service json
```
Comment (eldcn) [inline .github/workflows/android.yml:29]: In general, we can reuse steps by defining a yaml anchor before, e.g before jobs:

```yml
firebase-setup: &firebase-setup
  - name: Decode and write google service json
    env:
      GOOGLE_SERVICES_JSON: ${{ secrets.GOOGLE_SERVICES_JSON }}
    run: echo $GOOGLE_SERVICES_JSON | base64 --decode >./app/google-services.json
```

and reuse it here: `- <<: *firebase-setup` and in other jobs.

Also, we need to inject the json only for jobs that require a project build (`gradlew build`). Do dokka, lint and test depend on build? Maybe test yes, but not sure whether it is needed for the others

Comment (PSchmiedmayer) [reply .github/workflows/android.yml:29]: That's a really cool idea; I have not used that before.
Unfortunately, according to [this issue actions/runner/#1182](https://github.com/actions/runner/issues/1182) this syntax doesn't seem to be supported in the GitHub Action YAML.

Comment (Basler182) [reply .github/workflows/android.yml:29]: I had also assumed that only build and test needed the google json, but the github action only went through when I added the google-services.json at every single step

**Comment by PSchmiedmayer** [comment]
@Basler182 Let us know if you need any additional input here and how you can integrate this in some of the improvements on main if needed 👍

**Comment by Basler182** [comment]
> @Basler182 Let us know if you need any additional input here and how you can integrate this in some of the improvements on main if needed 👍

the file got deleted meanwhile,  i ll create a new PR

**Pull request closed without merging**